apps/aevm/src/aevm_ae_primops.erl

```diff
@@ -297,30 +312,36 @@ aens_call(?PRIM_CALL_AENS_REVOKE, _Value, Data, State) ->
 aens_call(PrimOp, _, _, _) ->
     {error, {illegal_aens_primop_call, PrimOp}}.
 
-aens_call_resolve(Data, State) ->
+aens_call_resolve(Data, Chain) ->
     [Name, Key, Type] = get_args([string, string, typerep], Data),
-    Callback = fun(API, ChainState) -> API:aens_resolve(Name, Key, Type, ChainState) end,
-    no_dynamic_gas(fun() -> query_chain(Callback, State) end).
+    Callback = fun(ChainAPI, ChainState) -> ChainAPI:aens_resolve(Name, Key, Type, ChainState) end,
+    no_dynamic_gas(fun() -> query_chain(Callback, Chain) end).
 
-aens_call_preclaim(Data, State) ->
+aens_call_preclaim(Data, #chain{api = API, state = State} = Chain) ->
     [Addr, CHash, Sign0] = get_args([word, word, sign_t()], Data),
-    Callback = fun(API, ChainState) -> API:aens_preclaim(<<Addr:256>>, <<CHash:256>>, to_sign(Sign0), ChainState) end,
-    no_dynamic_gas(fun() -> cast_chain(Callback, State) end).
+    {ok, Tx} = API:aens_preclaim_tx(<<Addr:256>>, <<CHash:256>>, State),
+    SizeGas = size_gas(Tx),
+    Callback = fun(ChainAPI, ChainState) -> ChainAPI:aens_preclaim(Tx, to_sign(Sign0), ChainState) end,
+    {ok, SizeGas, fun() -> cast_chain(Callback, Chain) end}.
```

Comment:
Doc update required to https://github.com/aeternity/protocol/blame/3bc313196b6e0b0319fa892e17d5bb9327c10395/contracts/aevm.md#L35-L42 (including AENS also).